protocol/x/perpetuals/keeper/perpetual.go

```diff
@@ -455,10 +455,32 @@ func (k Keeper) sampleAllPerpetuals(ctx sdk.Context) (
 		metrics.Latency,
 	)
 
+	marketIdToIndexPrice := k.pricesKeeper.GetMarketIdToValidIndexPrice(ctx)
+
 	for _, perp := range allPerpetuals {
-		marketPrice, err := k.pricesKeeper.GetMarketPrice(ctx, perp.Params.MarketId)
-		if err != nil {
-			panic(err)
+		indexPrice, exists := marketIdToIndexPrice[perp.Params.MarketId]
+		// If a valid index price is missing, skip this market, effectively emitting a zero premium.
+		if !exists {
+			k.Logger(ctx).Error(
```

Comment:
As a follow-up task, you may want to escalate this message from INFO to ERROR if it fires a lot on network deploy / reset.